libr/debug/p/debug_native.c

```diff
@@ -660,15 +660,24 @@ static int bsd_reg_read(RDebug *dbg, int type, ut8* buf, int size) {
 #endif
 		return true;
 		break;
+	
 	case R_REG_TYPE_FPU:
+	case R_REG_TYPE_VEC64: // MMX
 	   ret = ptrace (PT_GETFPREGS, pid, (caddr_t)buf, sizeof (struct fpreg));
 	   break;
-	case R_REG_TYPE_VEC64: // MMX
 	case R_REG_TYPE_VEC128: // XMM
 	case R_REG_TYPE_VEC256: // YMM
 	case R_REG_TYPE_VEC512: // ZMM
-		// not implemented
-		break;
+	      #if __KFBSD__
+		struct ptrace_xstate_info info;
+               ret = ptrace(PT_GETXSTATE_INFO, pid, (caddr_t)&info, sizeof(info));
+               
+               if (info.xsave_len != 0)
+               {
+                ret = ptrace(PT_GETXSTATE, pid,  (caddr_t)buf, info.xsave_len);
+               }
+              #endif
```

Comment:
```suggestion
              #endif
```

dont indent preprocessor directives